src/address_mapper.ts

```diff
@@ -49,19 +50,22 @@ export class AddressMapper {
     )
   }
 
-  private _signIdentityMapping(from: Address, to: Address, privateKey: Uint8Array) {
-    const hash = keccak256([from.toString(), to.toString()])
-    return ecsign(hash, privateKey)
-  }
-
-  async addIdentityMapping(from: Address, to: Address, privateKey: Uint8Array) {
+  async addIdentityMapping(from: Address, to: Address, web3: Web3, ethAddress: string) {
```

Comment:
A `Web3Signer` instance should be passed in as a parameter, you shouldn't create it in here.

Reply:
Also this function should be named `addIdentityMappingAsync`